buildtools/commands/test.js

```diff
@@ -0,0 +1,74 @@
+const { resolve } = require('path');
+const { execSync } = require('child_process');
+
+exports.command = 'test';
+exports.desc = 'Runs tests for the Okta Sign-in Widget';
+exports.builder = {
+  type: {
+    description: 'Type of the test runner',
+    choices: [
+      'karma',
+      'jest',
+      'testcafe',
+    ],
+    required: true,
+  },
+  config: {
+    description: 'Optional override for the test config file',
+  },
+  suiteHelp: {
+    description: 'Run this to see the test CLI options',
+  },
+};
+
+const suiteMap = {
+  karma: {
+    bin: 'karma start',
+    config: 'karma.conf.js',
+    preReq: [
+      'grunt assets',
+    ],
+  },
+  jest: {
+    bin: 'jest',
+    config: 'jest.config.js',
+    preReq: [
+      'grunt assets',
```

Comment:
I only briefly looked into migrating `testcafe`, which has different `preReq`s that jest/karma. Moving the required setup scripts to an object like this seemed like the best solution.